qiime2/core/type/primitive.py

```diff
@@ -188,6 +188,10 @@ def to_ast(self):
 
 class Choices(Predicate):
     def __init__(self, choices):
+
+        if not choices:
+            raise ValueError("'Choices' cannot be instantiated with an empty set.")
```

Comment:
[Looks like `flake8` found an issue](https://travis-ci.org/qiime2/qiime2/builds/291293905?utm_source=github_status&utm_medium=notification). It checks compliance with [PEP 8](https://www.python.org/dev/peps/pep-0008/). We use style guides like these because they can be automatically checked, and it prevents fights over formatting.

In this case you can use Python's implicit string concatenation:
```python
raise ValueError("'Choices' cannot be instantiated with an empty"
                 " set.")
```

Python will mash those together because they are part of the same expression. Just be sure to mind the space (` `) otherwise you'll have two words mashed together as well.

You can also check locally by running `make lint`.